File: Practice/25/С++/Project/Project.cpp
#include <iostream>
#include <vector>
#include <random>

std::vector<int> BozoSort(std::vector<int> Num, bool small = true);
std::vector<int> BozoSort(std::vector<std::vector<int>> Num, bool small = true);
std::vector<int> BozoSort(int a, int b, int c, bool small = true);
void VectorPrnt(std::vector<int> Num);

int main()
{
    int n;
    std::cin >> n;
    std::vector<int> Num(n);
    std::vector<int> Res;
    for (int i = 0; i < n; i++) {
        std::cin >> Num[i];
    }

    std::vector<std::vector<int>> ARR;
    std::vector<int> buf;
    if (n % (int)sqrt(n) == 0) {
        for (int i = 0; i < n; i++) {
            buf.push_back(Num[i]);
            if (((i + 1) % int(sqrt(n))) == 0) {
                ARR.push_back(buf);
                buf.resize(0);
            }
        }
    }

    VectorPrnt(BozoSort(Num));
    VectorPrnt(BozoSort(Num, false));

    VectorPrnt(BozoSort(ARR));
    VectorPrnt(BozoSort(ARR, false));

    VectorPrnt(BozoSort(Num[0], Num[1], Num[2]));
    VectorPrnt(BozoSort(Num[0], Num[1], Num[2], false));
}

void VectorPrnt(std::vector<int> Num) {
    for (auto i = Num.begin(); i < Num.end(); ++i) {
        std::cout << *i << " ";
    }
    std::cout << "\n";
}

std::vector<int> BozoSort(std::vector<int> Num, bool small) {
    while (true)
    {
        int n1 = 0 + rand() % int(Num.size());
        int n2 = 0 + rand() % int(Num.size());
        if (n1 != n2) std::swap(Num[n1], Num[n2]);
        else continue;
        if (small == true) {
            int res = 0;
            for (int i = 0; i < (int(Num.size())-1); i++) {
                if (Num[i] < Num[i + 1]) res++;
                else break;
                if (res == int(Num.size()-1)) {
                    return Num;
                }
            }
        }
        else {
            int res = 0;
            for (int i = 0; i < (int(Num.size()) - 1); i++) {
                if (Num[i] > Num[i + 1]) res++;
                else break;
                if (res == int(Num.size() - 1)) {
                    return Num;
                }
            }
        }
    }
}

std::vector<int> BozoSort(std::vector<std::vector<int>> Num, bool small) {

    std::vector<int> buf;
    
    for (int i = 0; i < int(Num.size()); i++) {
        for (int j = 0; j < int(Num.size()); j++) {
            buf.push_back(Num[i][j]);
        }
    }
    if (small == true) {
        buf = BozoSort(buf, true);
    }
    else {
        buf = BozoSort(buf, false);
    }
    return buf;
}

std::vector<int> BozoSort(int a, int b, int c, bool small) {
    std::vector<int> buf;
    buf.push_back(a);
    buf.push_back(b);
    buf.push_back(c);

    if (small == true) buf = BozoSort(buf, true);
    else buf = BozoSort(buf, false);

    return buf;
}

File: Practice/25/Python/Project.py
import random
import math

def ArrPrint(Arr):
    for x in Arr: print(x, end = ' ')
    print()
    
def swap(v, x, y):
    t = v[x]
    v[x] = v[y]
    v[y] = t
    
def isSorted(LST, small):
    if small == True:
        for i in range(1, len(LST)):
            if(LST[i - 1] > LST[i]):
                return False
        return True
    else:
        for i in range(1, len(LST)):
            if(LST[i - 1] < LST[i]):
                return False
        return True
        
def BozoSort(a, b = None , c = None, small = True):
    def Bozo(num, small):
        while True:
                n1 = random.randint(0, (len(num)-1))
                n2 = random.randint(0, (len(num)-1))
                if (n1 != n2):
                    swap(num, n1, n2)
                else:
                    continue;
                if isSorted(num, small):
                    return num
    if b == None:
        if isinstance(a[0], int):
            return Bozo(a, small)
        elif isinstance(a[0], list):
            i = 0
            j = 0
            buf = []
            while i < len(a):
                buf.extend(a[i])
                i += 1;
            buf = Bozo(buf, small)
            return buf
    else:
        buf = []
        buf.append(a)
        buf.append(b)
        buf.append(c)
        buf = Bozo(buf, small)
        return buf;

b = input()
c = input()
d = c.split()
D = [int(x) for x in d]
if int(b) == len(D):
    D1 = D
    ARR = []
    buf = []
    for i in D:
        buf.append(i)
        if math.sqrt(int(b)) == len(buf):
            ARR.append(buf)
            buf = []
            
    ArrPrint(BozoSort(D))
    ArrPrint(BozoSort(D, None, None,False))
    ArrPrint(BozoSort(ARR))
    ArrPrint(BozoSort(ARR, None, None, False))
    ArrPrint(BozoSort(D1[0], D1[1], D1[2]))
    ArrPrint(BozoSort(D1[0], D1[1], D1[2], False))
else:
    exit()